### PR TITLE
style(#1874): resolve typo function name for data handler

### LINKED ISSUE #1874: typo method name from `DataHandler` abstract method
https://github.com/4paradigm/OpenMLDB/blob/6f993f3dea4b0b1aa12c487eb110f8b42ff4e5db/hybridse/include/vm/catalog.h#L89-L89

should be `GetHandlerType`.

To fix this, you need change the method name for both abstract class and inherit classes, like:
+ https://github.com/4paradigm/OpenMLDB/blob/f83616596240bef0f30ba0e0d9069ed45f42ebde/hybridse/include/vm/catalog.h#L170
+ https://github.com/4paradigm/OpenMLDB/blob/f83616596240bef0f30ba0e0d9069ed45f42ebde/hybridse/include/vm/catalog.h#L232

then update callee names elsewhere as well

### PR DESCRIPTION
**Pull request by PrajwalBorkar**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Fixed #1874 

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: hybridse/include/vm/catalog.h [hunk 1]
```diff
@@ -358,7 +358,7 @@ class PartitionHandler : public TableHandler {
     virtual std::unique_ptr<WindowIterator> GetWindowIterator() = 0;
 
     /// Return HandlerType::kPartitionHandler by default
-    const HandlerType GetHanlderType() override { return kPartitionHandler; }
+    const HandlerType GetHandlerType() override { return kPartitionHandler; }
```
Comment (aceforeverd) [inline hybridse/include/vm/catalog.h:361]: pls update method calling as well

**Review by aceforeverd — CHANGES_REQUESTED** [review]

File: hybridse/include/vm/catalog.h (hunk 1 above)
Comment (PrajwalBorkar) [reply hybridse/include/vm/catalog.h:361]: Can u explain in which file is calling method?

Comment (aceforeverd) [reply hybridse/include/vm/catalog.h:361]: Screen Shot 2022-05-25 at 5 18 38 PM

**Review by aceforeverd — APPROVED** [review]

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Review by dl239 — APPROVED** [review]

**Pull request merged**